modules/ds18b20/ds18b20.h

```diff
@@ -0,0 +1,281 @@
+ * @{
+ *
+ * With this you can read temperature, set and get temperature resolution from 9 to 12 bits and check if device is DS18B20.
+ * 
+ * Pin for STM32Fxxx is the same as set with @ref ONEWIRE library.
+ *
+ * \par Changelog
+ *
+\verbatim
+ Version 1.0
+  - First release
+\endverbatim
+ *
+ * \par Dependencies
+ *
+\verbatim
+ - STM32Fxxx HAL
+ - TM ONEWIRE
+ - TM GPIO
+ - defines.h
+\endverbatim
+ */
+
+#include <modules/onewire/onewire.h>
+
+/**
+ * @defgroup DS18B20_Macros
+ * @brief    Library defines
+ * @{
+ */
+
+/* More details: https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf */
+
+/* Every onewire chip has different ROM code, but all the same chips has same family code */
+/* in case of DS18B20 this is 0x28 and this is first byte of ROM address */
+#define DS18B20_FAMILY_CODE             0x28
+/* DS18B20 Commands */
+#define DS18B20_CMD_ALARMSEARCH         0xEC
+#define DS18B20_CMD_CONVERTTEMP         0x44    /* Convert temperature */
+
+/* DS18B20 Resolution defines */
+#define DS18B20_DECIMAL_STEPS_12BIT     0.0625
+#define DS18B20_DECIMAL_STEPS_11BIT     0.125
+#define DS18B20_DECIMAL_STEPS_10BIT     0.25
+#define DS18B20_DECIMAL_STEPS_9BIT      0.5
+
+#define DS18B20_READ_TIMEOUT_MS         2000  /* Large timeout in ms for any read. Max read time should be ~1000ms for 12bit resolution */
+
+/* Bits locations for resolution */
+#define DS18B20_RESOLUTION_R1           6
+#define DS18B20_RESOLUTION_R0           5
+
+/* DS18B20 Common Numbers */
+#define DS18B20_READ_DATA_SIZE          9
+#define DS18B20_READ_CRC_BYTE           8
+#define DS18B20_DATA_LSB                0
+#define DS18B20_DATA_MSB                1
+
+#define DS18B20_USE_CRC 1
+
+/* DS18B20 Common Numbers */
+#define DS18B20_DATA_CRC_BYTE  8
+#define DS18B20_DATA_LSB       0
+#define DS18B20_DATA_MSB       1
+
+#define DS18B20_MAX_TEMP_DEG_C 125
+#define DS18B20_MIN_TEMP_DEG_C (-55)
+
+#define DS18B20_CONFIG_REGISTER_BYTE          4
+#define DS18B20_CONFIG_REGISTER_R0_R1_BITMASK 0x60
+#define DS18B20_CONFIG_REGISTER_RESERVED_BITS 5
+
+#define DS18B20_TEMP_SIGN_BITMASK             0x8000/**
+ * @}
+ */
+
+/**
+ * @defgroup DS18B20_Typedefs
+ * @brief    Library Typedefs
+ * @{
+ */
+
+/**
+ * @brief  DS18B0 Resolutions available
+ */
+typedef enum {
+    DS18B20_RESOLUTION_9BITS  = 9,  /*!< DS18B20 9 bits resolution */
+    DS18B20_RESOLUTION_10BITS = 10, /*!< DS18B20 10 bits resolution */
+    DS18B20_RESOLUTION_11BITS = 11, /*!< DS18B20 11 bits resolution */
+    DS18B20_RESOLUTION_12BITS = 12  /*!< DS18B20 12 bits resolution */
+} DS18B20_Resolution_t;
+
+/**
+ * @brief  DS18B0 Return Codes
+ */
+typedef enum {
+    DS18B20_USAGE_ERROR            = -5,  /*!< Function Usage Error, check parameters */
+    DEVICE_NOT_DS18B20             = -4,  /*!< Device does not match DS18B20 Family Code */

```

Comment:
can we rename DEVICE_NOT_DS18B20 to DS18B20_INVALID_DEVICE?  Mainly keep the same DS18B20 prefix to match the rest of the others since C doesn't have namespace enforcement.